src/Routes/MoviePage/index.js

```diff
@@ -0,0 +1,68 @@
+import { MovieInfo, Cast, Crew, CastHeader, CrewHeader, CastList, CrewList } from "./styled";
+import MovieCardFull from "../../common/MovieCardFull"
+import MovieHeading from "./components/MovieHeading.js"
+import { useParams } from "react-router-dom";
+import { useDataById } from "../../useDataById";
+import { useSelector } from "react-redux";
+import { selectMovies, selectCast, selectCrew } from "../../moviesSlice";
+import { useMovieCast } from "../../useMovieCast";
+import PersonCard from "../../common/PersonCard";
+
+const MoviePage = () => {
+    const params = useParams();
+
+    useDataById(params.id);
+    useMovieCast(params.id);
+
+    const movie = useSelector(selectMovies);
+    const cast = useSelector(selectCast);
+    const crew = useSelector(selectCrew);
+
+    // these if statements are kinda bad but i didnt know any other way of making operations on selected variables :C
+
+    let rating = 0;
+    if (movie.vote_average) {
+        rating = movie.vote_average.toFixed(1);
+    }
+
+    let tags = [];
+    if (movie.genres) {
+        movie.genres.forEach(genre => {
+            tags.push(genre.name)
+        })
+    }
+
+    let production = "";
```

Comment:
Network suggests  to use  ` .join(', ')  ` method but I never used that. 
Probably it can looks like that 
`
  const production = movie.production_companies
        ? movie.production_companies.map(company => company.name).join(', ')
        : "";  
 `
        but i'm not sure 
        if you want try, just try if you don't want, don't do that ;)